prometheus/prometheus.yml

```diff
@@ -0,0 +1,9 @@
+global:
+  scrape_interval: 5s
+
+scrape_configs:
+
+  - job_name: 'myapp'
+    metrics_path: '/v2/metrics'
+    static_configs:
+      - targets: ['172.17.0.1:8080']
```

Comment:
- targets: ['host.docker.internal:8080']

Reply:
I found a mistake in prometheus/prometheus.yml. It looks like the target IP is outdated. I think you should change the following

Reply:
As i understand the `172.17.0.1` refers to the IP address of the default gateway on the default bridge network and is not outdated. However the `host.docker.internal` is used in Docker desktop and won't work on Linux systems without additional configuration.
If we want to support Windows and Linux together we can add the following to the docker-compose.yaml for prometheus
```yaml
    extra_hosts:
      - "host.docker.internal:host-gateway"
```
and use `host.docker.internal` as our targets in prometheus/prometheus.yml
What do you think? 
@Iajrdev

Reply:
Thank you for completing this. I think this configuration is okay for Linux, macOS, and Windows OS:

[prometheus/prometheus.yml](https://github.com/PwPJ/linkedlist/pull/11/files/77626bd694fede62db808a8e03fc0872f3a42a66#diff-79c19119fd226b897b5aa398c6623a078e9ae00fbd2fe20807e1fbb373fa0294)
```diff
+     - targets: ['host.docker.internal:8080']
```

[docker-compose.yml](https://github.com/PwPJ/linkedlist/pull/11/files/77626bd694fede62db808a8e03fc0872f3a42a66#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3)

```diff
+  extra_hosts:
+    - "host.docker.internal:host-gateway"
```

Reply:
To clarify, you're suggesting that we should remove `'172.17.0.1:8080'` from the `prometheus.yml` file and replace it with `'host.docker.internal:8080'`?
@alipourhabibi @armineyvazi

Reply:
To clarify, the suggestion is to replace `172.17.0.1:8080` with `host.docker.internal:8080` in the `prometheus.yml` file. Additionally, we should add the following line to the `docker-compose.yaml` file to ensure compatibility across both Windows/Mac and Linux environments:

```yaml
extra_hosts:
  - "host.docker.internal:host-gateway"
```

This configuration will make `host.docker.internal` resolvable within Docker containers on all platforms.

Reply:
I have made changes to the files as per your review and pushed them.
Please review them again.

Reply:
I am not sure, but I think it should look like this, not added in Grafana:

```diff

  prometheus:
    image: docker.arvancloud.ir/prom/prometheus:latest
    container_name: prometheus
    volumes:
      - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml
      - prometheus_data:/prometheus
    ports:
      - "9090:9090"
+  extra_hosts:
+   - "host.docker.internal:host-gateway"
    restart: unless-stopped

```

If you have a Linux system, please test this configuration to ensure it works correctly. The extra_hosts directive ensures that host.docker.internal is resolvable within the Prometheus container. Add the Grafana service if needed, without extra_hosts unless required for Grafana.

@Iajrdev